### PR TITLE
PDF representation CLIs

### PR DESCRIPTION
**Pull request by cgsandford**
Please see JIRA ticket for description of changes

Associated suite PR (simultaneous): https://github.com/MetOffice/improver_suite/pull/556

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           

Complexity increasing per file
==============================
- improver_tests/acceptance/test_generate_realizations.py  2
- improver/cli/generate_percentiles.py  8
- improver_tests/acceptance/test_generate_percentiles.py  1
- improver/cli/generate_realizations.py  8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5435102&bid=17823440)

File: improver_tests/metadata/test_probabilistic.py [hunk 1]
```diff
@@ -135,6 +137,30 @@ def test_error_not_probability(self):
             extract_diagnostic_name("lwe_precipitation_rate")
 
 
+def Test_is_probability(IrisTest):
```
Comment (benfitzpatrick) [inline improver_tests/metadata/test_probabilistic.py:140]: Codacy Issue found: [Argument name "IrisTest" doesn't conform to snake_case naming style](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5435102)

Comment (benfitzpatrick) [inline improver_tests/metadata/test_probabilistic.py:140]: Codacy Issue found: [Redefining name 'IrisTest' from outer scope (line 38)](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5435102)

File: improver_tests/metadata/test_probabilistic.py [hunk 2]
```diff
@@ -135,6 +137,30 @@ def test_error_not_probability(self):
...
+        self.data = np.ones((3, 3, 3), dtype=np.float32)
+        self.threshold_points = np.array([276, 277, 278], dtype=np.float32)
+
+    def test_true(self):
+        """Test a probability cube evaluates as true"""
+        cube = set_up_probability_cube(self.data, self.threshold_points)
+        result = is_probability(cube)
+        self.assertTrue(result)
+
+    def test_false(self):
```
Comment (benfitzpatrick) [inline improver_tests/metadata/test_probabilistic.py:154]: Codacy Issue found: [Unused variable 'test_false'](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5435102)

File: improver_tests/metadata/test_probabilistic.py [hunk 3]
```diff
@@ -135,6 +137,30 @@ def test_error_not_probability(self):
             extract_diagnostic_name("lwe_precipitation_rate")
 
 
+def Test_is_probability(IrisTest):
+    """Test the is_probability function"""
+
+    def setUp(self):
```
Comment (benfitzpatrick) [inline improver_tests/metadata/test_probabilistic.py:143]: Codacy Issue found: [Unused variable 'setUp'](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5435102)

File: improver_tests/metadata/test_probabilistic.py [hunk 4]
```diff
@@ -135,6 +137,30 @@ def test_error_not_probability(self):
...
+        """Set up test data"""
+        self.data = np.ones((3, 3, 3), dtype=np.float32)
+        self.threshold_points = np.array([276, 277, 278], dtype=np.float32)
+
+    def test_true(self):
```
Comment (benfitzpatrick) [inline improver_tests/metadata/test_probabilistic.py:148]: Codacy Issue found: [Unused variable 'test_true'](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5435102)

File: improver_tests/metadata/test_probabilistic.py (hunk 1 above)
Comment (benfitzpatrick) [inline improver_tests/metadata/test_probabilistic.py:140]: Codacy Issue found: [Unused argument 'IrisTest'](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5435102)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver/cli/generate_realizations.py  8
- improver_tests/acceptance/test_generate_percentiles.py  1
- improver/cli/generate_percentiles.py  8
- improver_tests/acceptance/test_generate_realizations.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5435102&bid=17823440)

File: improver/cli/generate_realizations.py [hunk 5]
```diff
@@ -57,40 +56,47 @@ def process(
...
+    from improver.metadata.probabilistic import is_probability
+    from improver.ensemble_copula_coupling.ensemble_copula_coupling import (
+        ConvertProbabilitiesToPercentiles,
+        RebadgePercentilesAsRealizations,
+        ResamplePercentiles,
+        EnsembleReordering,
+    )
+
+    if cube.coords("realization"):
+        return cube
```
[the remark this answers is not in the document]
Comment (cgsandford) [reply improver/cli/generate_realizations.py:77]: This doesn't change what the previous CLI did - it's just by putting a separate `return` statement here I can avoid a bit of cyclometric complexity (couple of layers of `if / else`s) further down.  The previous CLI just had `if percentiles ... elif probabilities ... elif realizations [result = cube] ... else [raise error] ... return result`.

Is there something else we would want to happen here?

File: improver_tests/acceptance/test_generate_realizations.py [hunk 6]
```diff
@@ -115,9 +113,19 @@ def test_realizations(tmp_path):
     acc.compare(output_path, input_path)
 
 
+def test_error_no_realizations_count(tmp_path):
```
Comment (benfitzpatrick) [inline improver_tests/acceptance/test_generate_realizations.py:116]: Codacy Issue found: [Function name "test_error_no_realizations_count" doesn't conform to '[a-z_][a-z0-9_]{2,30}$' pattern](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5435102)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- improver_tests/acceptance/test_generate_realizations.py  2
- improver_tests/acceptance/test_generate_percentiles.py  1
- improver/cli/generate_realizations.py  8
- improver/cli/generate_percentiles.py  8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5435102&bid=17823440)

File: improver_tests/acceptance/test_generate_percentiles.py [hunk 7]
```diff
@@ -28,19 +28,19 @@
...
-"""Tests for the percentile CLI"""
+"""Tests for the generate-percentiles CLI"""
 
 import pytest
 
 from . import acceptance as acc
 
 pytestmark = [pytest.mark.acc, acc.skip_if_kgo_missing]
-run_cli = acc.run_cli("percentile")
+run_cli = acc.run_cli("generate-percentiles")
```
Comment (benfitzpatrick) [inline improver_tests/acceptance/test_generate_percentiles.py:38]: Codacy Issue found: [Constant name "run_cli" doesn't conform to '(([A-Z_][A-Z0-9_]*)|(__.*__))$' pattern](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5435102)

File: improver_tests/metadata/test_probabilistic.py [hunk 8]
```diff
@@ -135,6 +137,30 @@ def test_error_not_probability(self):
             extract_diagnostic_name("lwe_precipitation_rate")
 
 
+class Test_is_probability(IrisTest):
```
Comment (benfitzpatrick) [inline improver_tests/metadata/test_probabilistic.py:140]: Codacy Issue found: [Class name "Test_is_probability" doesn't conform to '[A-Z_][a-zA-Z0-9]+$' pattern](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5435102)

File: improver/cli/generate_realizations.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (bayliffe) [reply improver/cli/generate_realizations.py:77]: No I don't think so, it was more a note to myself that the case of percentiles and probabilities in the same cube is not handled (beyond simply not converting), but that I think this is okay.

File: improver_tests/metadata/test_probabilistic.py [hunk 9]
```diff
@@ -135,6 +137,30 @@ def test_error_not_probability(self):
...
+    """Test the is_probability function"""
```
Comment (MoseleyS) [inline improver_tests/metadata/test_probabilistic.py:141]: Please could you add a test for a cube with a scalar threshold coord?

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver/cli/generate_realizations.py  8
- improver/cli/generate_percentiles.py  8
- improver_tests/acceptance/test_generate_realizations.py  2
- improver_tests/acceptance/test_generate_percentiles.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5435102&bid=17823440)

**Pull request merged**